File: 151004/Sakovsky/src/marker/marker.controller.ts
import { Body, Controller, Delete, Get, HttpCode, NotFoundException, Param, ParseIntPipe, Post, Put, UsePipes, ValidationPipe } from '@nestjs/common';
import { MarkerRequestToCreate } from '../dto/request/MarkerRequestToCreate';
import { MarkerResponseTo } from '../dto/response/MarkerResponseTo';
import { MarkerService } from './marker.service';
import { MarkerRequestToUpdate } from '../dto/request/MarkerRequestToUpdate';

@Controller('markers')
export class MarkerController {
    constructor(private readonly markerService: MarkerService ){}

    @Get()
    async getAll(): Promise<MarkerResponseTo[]>{
        const result: MarkerResponseTo[] = [];
        const markers = await this.markerService.getAll();
        for (const marker of markers) {
            const markerDto = new MarkerResponseTo();
            markerDto.id = marker.id;
            markerDto.name = marker.name;
            result.push(markerDto);
        }
        return result;
    }

    @Get(':id')
    async getById(@Param('id', ParseIntPipe) id: number): Promise<MarkerResponseTo>{
        const markerDto = new MarkerResponseTo();
        const marker = await this.markerService.getById(id);
        if(!marker){
            throw new NotFoundException('Marker not found');
        }
        markerDto.name = marker.name;
        markerDto.id = marker.id;
        return markerDto;
    }

    @UsePipes(new ValidationPipe())
    @Post()
    async createMarker(@Body() dto: MarkerRequestToCreate): Promise<MarkerResponseTo>{
        const markerDto = new MarkerResponseTo();
        const marker =  await this.markerService.createMarker(dto);
        markerDto.name = marker.name;
        markerDto.id = marker.id;
        return markerDto;
    }

    @UsePipes(new ValidationPipe())
    @Put()
    async updateMarker(@Body() authorDto: MarkerRequestToUpdate): Promise<MarkerResponseTo>{
        const markerDto = new MarkerResponseTo();
        const marker =  await this.markerService.updateMarker(authorDto);
        markerDto.name = marker.name;
        markerDto.id = marker.id;
        return markerDto;
    }

    @HttpCode(204)
    @Delete(':id')
    deleteMarker(@Param('id', ParseIntPipe) id: number): Promise<void>{
        return this.markerService.deleteById(id);
    }
}

File: 151002/Strukov/REST News/src/main/java/by/bsuir/news/controller/MarkerController.java
package by.bsuir.news.controller;

import by.bsuir.news.dto.request.MarkerRequestTo;
import by.bsuir.news.dto.response.MarkerResponseTo;
import by.bsuir.news.entity.Marker;
import by.bsuir.news.repository.MarkerRepository;
import by.bsuir.news.service.GenericService;
import by.bsuir.news.service.MarkerService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/markers")
public class MarkerController {
    @Autowired
    private MarkerService markerService = new MarkerService();

    @GetMapping
    public ResponseEntity getAllMarkers() {
        try {
            return new ResponseEntity(markerService.getAll(), HttpStatus.OK);
        } catch (Exception e) {
            return ResponseEntity.badRequest().body(null);
        }
    }

    @GetMapping("/{id}")
    public ResponseEntity getMarker(@PathVariable Long id) {
        try {
            return new ResponseEntity(markerService.getById(id), HttpStatus.OK);
        } catch (Exception e) {
            return ResponseEntity.badRequest().body(id);
        }
    }

    @PostMapping
    public ResponseEntity saveMarker(@RequestBody MarkerRequestTo marker) {
        try {
            return new ResponseEntity(markerService.create(marker), HttpStatus.CREATED);
        } catch (Exception e) {
            return ResponseEntity.badRequest().header(e.getMessage()).body(marker);
        }
    }

    @PutMapping("")
    public ResponseEntity updateMarker(@RequestBody MarkerRequestTo marker) {
        try {
            return new ResponseEntity(markerService.update(marker), HttpStatus.OK);
        } catch (Exception e) {
            return ResponseEntity.badRequest().header(e.getMessage()).body(null);
        }
    }

    @DeleteMapping("/{id}")
    public ResponseEntity deleteMarker(@PathVariable Long id) {
        try {
            return new ResponseEntity(markerService.delete(id), HttpStatus.NO_CONTENT);
        } catch (Exception e) {
            return ResponseEntity.badRequest().body(null);
        }
    }
}
